File: 70_climbing_stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        a, b = 1, 2  # a-> # of steps for previous stair, b-> # of steps for current stair
        if n == 1: return a
        if n == 2: return b
        for _ in range(3, n+1):
            a, b = b, a+b
            
        return b

File: 70_climbing_stairs.java
class Solution {
    public int climbStairs(int n) {
        if(n == 1) return 1;
        if(n == 2) return 2;
        int i1 = 1, i2 = 2, stair = 3;

        while(stair <= n){
            int temp = i2;
            i2 = i1 + i2;
            i1 = temp;
            stair++;
        }
        
        return i2;
    }
}
